project007/webapp/controller/ListReport.controller.js

```diff
@@ -271,6 +242,42 @@ sap.ui.define(
           }
         );
       },
+
+      /**
+      * Combine all filters.
+      */
+       combineFilters: function() {
+        var oStateModel         = this.getView().getModel("stateModel");
+        var sAllFieldValue      = oStateModel.getProperty("/AllField")?.trim();
+        var aCategoryNameValue  = this._getArrayCategories();
+        var oItemsBinding       = this.byId("CategoriesTable").getBinding("items");
+        var aFilters            = [];
+
+        if (sAllFieldValue && !isNaN(sAllFieldValue)){
+          aFilters.push(new Filter("ID", FilterOperator.EQ, sAllFieldValue));
+        } else if (sAllFieldValue){
+          aFilters.push(new Filter("Name", FilterOperator.Contains, sAllFieldValue));
+        }
+
+        if (aCategoryNameValue) {
+          aFilters.push(...aCategoryNameValue.map((sValue) => {return new Filter("Name", FilterOperator.Contains, sValue)}));
+        }
+        oItemsBinding.filter(aFilters)
+      },
+
+      /**
+       * Get an array of suppliers.
+       * 
+       * @returns {Array} Suppliers.
+       * 
+       * @private
```

Comment:
add privacy in all functions:)

Reply:
@koryakovskaya I don't have enough salary=) For all private functions)